src/utils/gasOverrides.ts

```diff
@@ -0,0 +1,28 @@
+type GasOverrideOptions = {
+  base?: bigint;
+  percentIncrease?: bigint;
+};
+
+export type TransactionRequestGasOverrides = {
+  gasLimit?: GasOverrideOptions;
+};
+
+export type TransactionRequestRetryableGasOverrides = {
+  deposit?: GasOverrideOptions;
+};
+
+export function applyGasOverrides({
+  gasOverrides,
+  estimatedGas,
+  defaultGas,
+}: {
+  gasOverrides: GasOverrideOptions;
+  estimatedGas?: bigint;
+  defaultGas: bigint;
+}) {
+  const baseEstimatedGas = gasOverrides.base ?? estimatedGas ?? defaultGas;
+
+  return gasOverrides.percentIncrease
+    ? baseEstimatedGas + (baseEstimatedGas * gasOverrides.percentIncrease) / 100n
+    : baseEstimatedGas;
+}
```

Comment:
I think we can simplify this a bit and just have a pure function for percent increase:

```suggestion
export function applyPercentIncrease({
  base,
  percentIncrease = 0
}: {
  base: bigint;
  percentIncrease?: bigint
}) {
  return base + (base * percentIncrease) / 100n
}
```